lockfile/src/cyclonedx.rs

```diff
@@ -0,0 +1,313 @@
+use std::ffi::OsStr;
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::anyhow;
+use phylum_types::types::package::PackageType;
+use purl::GenericPurl;
+use serde::Deserialize;
+use thiserror::Error;
+
+use crate::{Package, PackageVersion, Parse, ThirdPartyVersion};
+
+// Define a custom error for unknown ecosystems.
+#[derive(Error, Debug)]
+#[error("Could not determine ecosystem")]
+struct UnknownEcosystem;
+
+// Define the generic trait for components.
+trait Component {
+    fn component_type(&self) -> &str;
+    fn name(&self) -> &str;
+    fn version(&self) -> &str;
+    fn scope(&self) -> Option<&str>;
+    fn purl(&self) -> Option<&str>;
+    fn components(&self) -> Option<&[Self]>
+    where
+        Self: Sized;
```

Comment:
It's probably easier to just add this requirement to the trait directly? That way it doesn't have to be specified every time during implementation.